### PR TITLE
Add support for adding and updating custom databases

### PR DESCRIPTION
**Pull request by lebovic**
Adds:
- Support for adding or updating custom databases

Removes:
- Tool specific min/max files – these don't seem to be aligned with the underlying tools anymore, and were a guard against opaque errors. Now that we have better error handling, these seem less relevant.

### CODE REVIEW

File: toolchest_client/api/output.py [hunk 1]
```diff
@@ -42,6 +44,14 @@ def set_s3_uri(self, s3_uri):
     def set_output_path(self, output_path):
         self.output_path = output_path
 
+    def set_database(self, database_name=None, database_version=None):
+        """Sets the database name and database version for ensuring versioning and reproducibility.
+
+        `database_version` increments when updating a database through the API.
```
Comment (jherr-dev) [inline toolchest_client/api/output.py:50]: Pending api comments addressing this

File: toolchest_client/tools/api.py [hunk 2]
```diff
@@ -219,27 +224,32 @@ def diamond_blastp(inputs, output_path=None, output_primary_name="out_file.tsv",
...
-
-          >>> import toolchest_client as toolchest
-          >>> toolchest.diamond_blastx(
-          ...     tool_args="",
-          ...     inputs="./path/to/input.fa",
-          ...     output_path="./path/to/output/",
-          ...     output_primary_name="out_file.tsv",
-          ... )
+    :param inputs: Path to a file that will be passed in as input. FASTA or FASTQ formats are supported (it may be
+gzip compressed)
```
Comment (jherr-dev) [inline toolchest_client/tools/api.py:231]: Indent

File: toolchest_client/tools/api.py [hunk 3]
```diff
@@ -628,3 +638,90 @@ def unicycler(output_path=None, read_one=None, read_two=None, long_reads=None, t
...
+    - the success status is
+    toolchest_client.api.status.COMPLETE ('complete') instead of
+    toolchest_client.api.status.READY_TO_TRANSFER_TO_CLIENT ('ready_to_transfer_to_client)
+    - is_async is True by default
+
+    Note that it may take 24-48 hours for the custom database to be ready for use.
+
+    :param database_path: Path or list of paths (local or S3) to be passed in as inputs.
+    :param tool: Toolchest tool with which you use the database (e.g. toolchest.tools.Kraken2).
+    :param database_name: Name of database to update.
```
Comment (jherr-dev) [inline toolchest_client/tools/api.py:656]: Consider user set version.

File: toolchest_client/tools/api.py [hunk 4]
```diff
@@ -628,3 +638,90 @@ def unicycler(output_path=None, read_one=None, read_two=None, long_reads=None, t
...
+        tool_args="",
+        custom_database_path=None,
+        max_inputs=1000,
+        **kwargs,
+    )
+    output = instance.run()
+    return output
+
+
+def add_database(database_path, tool, database_name, **kwargs):
```
Comment (jherr-dev) [inline toolchest_client/tools/api.py:686]: Consider user set version.

**Review by jherr-dev — APPROVED** [review]
LGTM other than some nits and options to consider.

File: toolchest_client/api/output.py (hunk 1 above)
Comment (lebovic) [reply toolchest_client/api/output.py:50]: Responding on API-side

File: .gitignore [hunk 5]
```diff
@@ -142,6 +142,3 @@ toolchest_client/demo.py
 
 # Default temporary directory for splitting input files
 temp_toolchest*
-
-# Default directories for local runs of integration tests
-test_*
```
Comment (lebovic) [inline .gitignore:147]: @bcai2 this also ignored test files :P

Comment (bcai2) [reply .gitignore:147]: That doesn't seem to be the case for me -- all `.py` test files are in `tests/test_*.py` form, which doesn't match. @lebovic

Comment (bcai2) [reply .gitignore:147]: Per our convo, let's add this back and modify this to `/test_*`

File: toolchest_client/tools/tool.py [hunk 6]
```diff
@@ -31,15 +31,15 @@
 
 class Tool:
     def __init__(self, tool_name, tool_version, tool_args,
-                 inputs, min_inputs, max_inputs=None, output_path=None,
+                 inputs, min_inputs=1, max_inputs=100, output_path=None,
```
Comment (bcai2) [inline toolchest_client/tools/tool.py:34]: Why `max_inputs=100` instead of `None`?

File: tests/test_database_update.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    open(test_file_two, "w").close()
+
+    output = toolchest.update_database(
+        database_path=test_dir,
+        tool=toolchest.tools.Kraken2,
+        database_name="standard",
+    )
+
+    assert output.database_name == "standard"
+    assert output.database_version
```
Comment (bcai2) [inline tests/test_database_update.py:47]: Are we expecting the value of the version to be an arbitrary or a specific value here?

**Review by bcai2 — APPROVED** [review]
Generally looks good if tested (but this will require testing)

**Pull request merged**